src/lib/storage/table.cpp

```diff
@@ -122,8 +132,29 @@ std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const {
 }
 
 void Table::compress_chunk(const ChunkID chunk_id) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto compressed_chunk = std::make_shared<Chunk>();
+  auto compressed_segments = std::vector<std::shared_ptr<AbstractSegment>>(column_count());
+  auto threads = std::vector<std::thread>(column_count());
```

Comment:
```suggestion
  const auto column_count = column_count();
  auto compressed_segments = std::vector<std::shared_ptr<AbstractSegment>>(column_count());
  auto threads = std::vector<std::thread>(column_count());
```
You call `column_count()` multiple times and also `threads.size()` inside a for-loop which is also equal to `column_count`. Maybe it would be nicer to create a const variable `column_count` to store this.